intake_stac/catalog.py

```diff
@@ -165,6 +167,57 @@ class StacCollection(StacCatalog):
     name = 'stac_catalog'
     _stac_cls = pystac.Collection
 
+    def get_asset(
+        self,
+        asset,
+        storage_options=None,
+        merge_asset_storage_options=True,
+        merge_asset_open_kwargs=True,
+        **kwargs,
+    ):
+        r"""
+        Get a datasource for a collection-level asset.
+
+        Parameters
+        ----------
+        asset : str, optional
+            The asset key to use if multiple Zarr assets are provided.
+        storage_options : dict, optional
+            Additional arguments for the backend fsspec filesystem.
+        merge_asset_storage_option : bool, default True
+            Whether to merge the storage options provided by the asset under the
+            ``xarray:storage_options`` key with `storage_options`.
+        **kwargs
+            Additional keyword options are provided to the loader, for example ``consolidated=True``
+            to pass to :meth:`xarray.open_zarr`.
+
+        Notes
+        -----
+        The Media Type of the asset will be used to determine how to load the data.
+
+        Returns
+        -------
+        DataSource
+            The dataset described by the asset loaded into a dask-backed object.
+        """
+        try:
+            asset_ = self._stac_obj.assets[asset]
+        except KeyError:
+            raise KeyError(
+                f'No asset named {asset}. Should be one of {list(self._stac_obj.assets)}'
+            ) from None
+
+        storage_options = storage_options or {}
+        if merge_asset_storage_options:
+            asset_storage_options = asset_.extra_fields.get('xarray:storage_options', {})
+            storage_options.update(asset_storage_options)
+
+        if merge_asset_open_kwargs:
+            asset_open_kwargs = asset_.extra_fields.get('xarray:open_kwargs', {})
+            kwargs.update(asset_open_kwargs)
+
+        return StacAsset(asset, asset_)(storage_options=storage_options, **kwargs)
```

Comment:
took me a minute to figure out `asset` versus `asset_`.  should we use change the method parameter to `key` instead?